go/enclave/rollupchain/rollup_chain.go

```diff
@@ -201,38 +203,49 @@ func (rc *RollupChain) newBlockSubmissionResponse(bs *obscurocore.BlockState, ro
 //  STATE
 
 // Recursively calculates and stores the block state, receipts and logs for the given block.
-func (rc *RollupChain) updateState(b *types.Block) *obscurocore.BlockState {
+func (rc *RollupChain) updateState(b *types.Block) (*obscurocore.BlockState, error) {
 	// This method is called recursively in case of re-orgs. Stop when state was calculated already.
-	blockState, found := rc.storage.FetchBlockState(b.Hash())
-	if found {
-		return blockState
+	blockState, err := rc.storage.FetchBlockState(b.Hash())
+	if err != nil {
```

Comment:
This ignores the err != nil && err == errutil.ErrNotFound is that expected ?
It's also the only condition where the next instructions are executed, otherwise, they return. Can a comment be added to explain the goal of the operations ?

Reply:
Restructured to be a bit clearer.

Reply:
Sugestion: `if err != nil && !errors.Is(err, errutil.ErrNotFound)` (no need to multi line)